src/Dependencies/Collections/SegmentedList`1.cs

```diff
@@ -135,12 +135,25 @@ public int Capacity
                 {
                     if (value > 0)
                     {
-                        var newItems = new SegmentedArray<T>(value);
-                        if (_size > 0)
-                        {
-                            SegmentedArray.Copy(_items, newItems, _size);
-                        }
-                        _items = newItems;
+                        // Rather than creating a copy of _items, instead reuse as much of it's data as possible.
+                        var segments = SegmentedCollectionsMarshal.AsSegments(_items);
+
+                        var oldSegmentCount = segments.Length;
+                        var newSegmentCount = (value + SegmentedArrayHelper.GetSegmentSize<T>() - 1) >> SegmentedArrayHelper.GetSegmentShift<T>();
+
+                        // Grow the array of segments, if necessary
+                        Array.Resize(ref segments, newSegmentCount);
+
+                        // Resize all segments to full segment size from the last old segment to the next to last
+                        // new segment.
+                        for (var i = oldSegmentCount > 0 ? oldSegmentCount - 1 : 0; i < newSegmentCount - 1; i++)
```

Comment:
💭 What about calling out the case of empty at the beginning of this block and just using `new SegmentedArray<T>` for it? It took me a bit to figure out how `oldSegmentCount - 1` could be a problem.

Reply:
Seems reasonable. It took me a while to figure out that oldSegmentCount -1 was a problem too.

Reply:
I also went ahead and reduced the nesting in here, that was muddling things up for me, at least.

Reply:
➡️ Now simplified